pgjdbc/src/main/java/org/postgresql/jdbc/TypeInfoCache.java

```diff
@@ -29,10 +29,14 @@
 
   // pgname (String) -> java.sql.Types (Integer)
   private Map<String, Integer> _pgNameToSQLType;
+  // oid -> java.sql.Types
+  private Map<Integer, Integer> _oidToSQLType;
```

Comment:
Would it make sense to create a composite class with sqlType, pgName, oid and java class?
This would allow to reduce the number of maps keyed by oid and pgName.

Reply:
I agree that there are likely a couple of places for improvement in the data structures using for caching. Rather than explore that in this PR, I think that's a good candidate for a subsequent patch.